File: src/Cryptofolio.Infrastructure/Entities/Holding.cs
using System.Text.Json.Serialization;

namespace Cryptofolio.Infrastructure.Entities
{
    /// <summary>
    /// Models an asset holding in a specific wallet.
    /// </summary>
    public class Holding
    {
        /// <summary>
        /// The id.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The wallet that holds the asset.
        /// </summary>
        public Wallet Wallet { get; set; }

        /// <summary>
        /// The asset held.
        /// </summary>
        public Asset Asset { get; set; }

        /// <summary>
        /// The quantity held.
        /// </summary>
        public decimal Qty { get; set; }

        /// <summary>
        /// The initial value in the wallet currency.
        /// It's the sum of all initial values of every transaction in the holding.
        /// </summary>
        [JsonPropertyName("initial_value")]
        public decimal InitialValue { get; set; }

        /// <summary>
        /// The current value in the wallet currency.
        /// </summary>
        [JsonPropertyName("current_value")]
        public decimal CurrentValue { get; set; }

        /// <summary>
        /// The change in percent.
        /// </summary>
        public decimal Change { get; set; }
    }
}

File: src/Cryptofolio.App/ClientApp/src/app/models/holding.ts
import { Asset } from "./asset";
import { Wallet } from "./wallet";

export class Holding {
  id: string;
  wallet: Wallet;
  asset: Asset;
  qty: number;
  initialValue: number;
  currentValue: number;
  change: number;

  constructor(properties?: any) {
    Object.assign(this, properties);
    this.wallet = new Wallet(properties.wallet);
    this.asset = new Asset(properties.asset);
    this.initialValue = properties["initial_value"];
    this.currentValue = properties["current_value"];
  }
}